spec/text-editor-spec.coffee

```diff
@@ -4018,3 +4018,71 @@ describe "TextEditor", ->
       editor.setPlaceholderText('OK')
       expect(handler).toHaveBeenCalledWith 'OK'
       expect(editor.getPlaceholderText()).toBe 'OK'
+
+  describe 'Gutters', ->
+    describe 'Add/Remove Gutters', ->
+      it 'can add a right gutter', ->
+        options = priority: 1
+        gutter = editor.addRightGutter 'test-gutter', options
+        expect(editor.getRightGutters().length).toBe 1
+        expect(editor.getRightGutters()[0]).toBe gutter
+
+      it 'can add a left gutter', ->
+        gutter = editor.addLeftGutter 'test-gutter'
+        expect(editor.getLeftGutters().length).toBe 1
+        expect(editor.getLeftGutters()[0]).toBe gutter
+
+      it 'will notify listeners when a gutter is removed', ->
+        mockListener = onGutterRemoved: null
+        editor.onDidRemoveGutter (gutter) ->
```

Comment:
It looked like the interface between the models and the view layer was the TextEditor, which is why I added this. `Gutter::onDidDestroy` could work too. Related to the point directly below.